src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const outputStates = [stateCopy];
+
+  for (const action of actions) {
+    let currentState = { ...outputStates[outputStates.length - 1] };
+
+    switch (action.type) {
+      case 'addProperties' :
+        currentState = Object.assign({}, currentState, action.extraData);
```

Comment:
```suggestion
       Object.assign(currentState, action.extraData);
```
The previous reviewer means that you don't need to assign it to the current state at all